src/multicast/network.c

```diff
@@ -77,25 +77,37 @@ static void AllocBuffer(uv_handle_t* handle, size_t suggestedSize, uv_buf_t* buf
     buf->base = malloc(buf->len);
 }
 
-static void OnMcastRx(uv_udp_t* handle, ssize_t nread, const uv_buf_t* buf, const struct sockaddr* addr, unsigned flags)
+static void OnMcastRx(uv_udp_t* handle, ssize_t nread, const uv_buf_t* buf, const struct sockaddr* addr,
+                      unsigned flags)
 {
     DPS_MulticastReceiver* receiver = (DPS_MulticastReceiver*)handle->data;
     DPS_NetEndpoint ep;
 
-    DPS_DBGPRINT("OnMcastRx\n");
+    if (nread == 0 && !addr) {
```

Comment:
It seems odd that this ever happens, is this just some weird libuv behavior? Maybe add a comment explaining why this isn't a socket error.

Reply:
yes this is libuv specific.  From http://docs.libuv.org/en/v1.x/udp.html:

The receive callback will be called with nread == 0 and addr == NULL when there is nothing to read, and with nread == 0 and addr != NULL when an empty UDP packet is received.